farm/modeling/adaptive_model.py

```diff
@@ -501,85 +503,16 @@ def get_language(self):
-                raise NotImplementedError
-
-            # add more info to config
-            self.language_model.model.config.id2label = {id: label for id, label in enumerate(self.prediction_heads[0].label_list)}
-            self.language_model.model.config.label2id = {label: id for id, label in enumerate(self.prediction_heads[0].label_list)}
-            self.language_model.model.config.finetuning_task = "text_classification"
-            self.language_model.model.config.language = self.language_model.language
-            self.language_model.model.config.num_labels = self.prediction_heads[0].num_labels
-
-            # init model
-            transformers_model = AutoModelForSequenceClassification.from_config(self.language_model.model.config)
-            # transfer weights for language model + prediction head
-            setattr(transformers_model, transformers_model.base_model_prefix, self.language_model.model)
-            transformers_model.classifier.load_state_dict(
-                self.prediction_heads[0].feed_forward.feed_forward[0].state_dict())
-        elif self.prediction_heads[0].model_type == "token_classification":
-            # add more info to config
-            self.language_model.model.config.id2label = {id: label for id, label in enumerate(self.prediction_heads[0].label_list)}
-            self.language_model.model.config.label2id = {label: id for id, label in enumerate(self.prediction_heads[0].label_list)}
-            self.language_model.model.config.finetuning_task = "token_classification"
-            self.language_model.model.config.language = self.language_model.language
-            self.language_model.model.config.num_labels = self.prediction_heads[0].num_labels
-
-            # init model
-            transformers_model = AutoModelForTokenClassification.from_config(self.language_model.model.config)
-            # transfer weights for language model + prediction head
-            setattr(transformers_model, transformers_model.base_model_prefix, self.language_model.model)
-            transformers_model.classifier.load_state_dict(
-                self.prediction_heads[0].feed_forward.feed_forward[0].state_dict())
-        else:
-            raise NotImplementedError(f"FARM -> Transformers conversion is not supported yet for"
-                                      f" prediction heads of type {self.prediction_heads[0].model_type}")
-        pass
+        """
+        Convert an adaptive model to huggingface's transformers format. Returns a list containing one model for each
+        prediction head.
 
-        return transformers_model
+        :return: List of huggingface transformers models.
+        """
+        return Converter.convert_to_transformers(self)
 
     @classmethod
-    def convert_from_transformers(cls, model_name_or_path, device, task_type, processor=None):
+    def convert_from_transformers(cls, model_name_or_path, device, task_type=None, processor=None):
```

Comment:
Why is task_type optional here?

Reply:
Because we are inferring the task type automatically if task_type is not set. (https://github.com/deepset-ai/FARM/blob/aa39d7824a2e62d545a78df306a4684c407891f9/farm/modeling/adaptive_model.py#L800)

Reply:
Oh ok, missed that :+1: